napari/resources/__init__.py

```diff
@@ -1,22 +1,149 @@
+        Whether to recompile and overwrite the resources.
+        By default, resources will be rebuilt if any of the following are True:
+            - the resources file does not already exist.
+            - ``overwrite`` argument is True
+            - the ``NAPARI_REBUILD_RESOURCES`` environmental variable is set
+    
+    Raises
+    ------
+    PermissionError
+        If we cannot write to any of the requested locations.
+    """
+    # the resources filename holds the napari version, Qt API, and QT version
+    version = version or f'{__version__}_{API}_{QT_VERSION}'
+    filename = f'_qt_resources_{version}.py'
+    tempfile = None
+    # see if we can write to the current napari/resources directory
+    target_file = _try_touch_file(join(abspath(dirname(__file__)), filename))
+    # if not, try to write to ~/.config/napari
+    if not target_file:
+        target_file = expanduser(join('~', '.config', 'napari', filename))
+        target_file = _try_touch_file(target_file)
+    # if that still doesn't work, create a temporary file.
+    if not target_file:
+        tempfile = NamedTemporaryFile(suffix='.py')
+        target_file = _try_touch_file(tempfile.name)
+    # if we can't even write a temporary file, we're probably screwed...
+    if not target_file:
+        raise PermissionError(
+            "Could not write qt_resources to disk. Please report this with a "
+            "description of your environment (pip freeze) at "
+            "https://github.com/napari/napari/issues"
+        )
+    overwrite = (
+        overwrite
+        or bool(environ.get('NAPARI_REBUILD_RESOURCES'))
+        or tempfile is not None
+    )
+    # this actually builds the res.qrc Qt resources file, and then from that
+    # autogenerates the python resources file that needs to be imported
+    # if the file already exists and overwrite is False, it will not be
+    # regenerated.
+    respath = build_pyqt_resources(fspath(target_file), overwrite=overwrite)
+    # import the python resources file and add to sys.modules
+    spec = spec_from_file_location("napari._qt_resources", respath)
+    module = module_from_spec(spec)
+    # important to add to sys.modules! otherwise segfault when function ends.
+    sys.modules[spec.name] = module
+    spec.loader.exec_module(module)
+    # if we had to create a tempfile in the process, we can close it now.
+    if tempfile:
+        tempfile.close()
+    return respath
+
+
+@lru_cache(maxsize=4)
+def get_stylesheet(extra: Optional[List[str]] = None) -> str:
+    """Combine all qss files into single (cached) style string.
+    
+    Note, this string may still have {{ template_variables }} that need to be
+    replaced using the :func:`napari.utils.theme.template` function.  (i.e. the
+    output of this function serves as the input of ``template()``)
+
+    Parameters
+    ----------
+    extra : list of str, optional
+        Additional qss files to include in stylesheet, by default None
```

Comment:
file paths? It's slightly ambiguous here whether you are expecting paths or the text from the files.